test/test_examples.py

```
@@ -442,6 +442,102 @@ def test_cuse(output_checker):
+    # before forwarding the calls to libc.
+    #
+    # The delay can be disabled at call-time by setting the TEST746_DELAY_DISABLE
+    # environment variable. e.g:
+    # TEST746_DELAY_DISABLE=1 python3 -m pytest test/test_examples.py::test_746
+
+    # set the path for the library to preload
+    preloadLib = base_cmdline + \
+        [ pjoin(basename, "test", "test746.so") ]
+    preloadLib = preloadLib[0]
+    #print("preloadLib: ", preloadLib)
+
+    fuseMNT = str(tmpdir)
+    #print("fuseMNT: ", fuseMNT)
+
+    # set the FUSE binary path and command
+    fuseCMD = base_cmdline + \
+        [ pjoin(basename, 'example', 'passthrough_fh'),
+        "-f", fuseMNT]
+    #print("fuseCMD: ", str(fuseCMD))
+
+    # start FUSE with our preload library
+    os.environ["LD_PRELOAD"] = preloadLib
+    fuseProcess = subprocess.Popen(fuseCMD,
+                                   stdout=output_checker.fd,
+                                   stderr=output_checker.fd)
+    os.environ["LD_PRELOAD"] = ""
+
+    leftoverFilesCount = -1
+    try:
+        wait_for_mount(fuseProcess, fuseMNT)
+
+        # use TemporaryDirectory so that it gets cleaned up automatically
+        tempDir = tempfile.TemporaryDirectory(dir=fuseMNT + "/tmp/")
+        tempDirPath = tempDir.name
+        #print("tempDir: " + tempDirPath);
+
+        ## test for the race condition
+
+        # create the test file
+        tempFile, tempFilePath = tempfile.mkstemp(dir=tempDir.name)
+        #print("tempFile: " + tempFilePath);
+
+        # close and delete it back-to-back. with the induced
+        # delays, these two FUSE OPs should execute in parallel
+        os.close(tempFile)
+        os.unlink(tempFilePath)
+
+        # check if any ".fuse_hidden" files are still there
+        tempDirFiles = os.listdir(tempDirPath)
+        #print(tempDirFiles)
+
+        leftoverFilesCount = len(tempDirFiles)
+
+        if leftoverFilesCount == 0:
```

You can replace all this with `assert os.listdir(tempDirPath) == []`, no need for any print output from tests

---

This test started as a standalone test file before merging with `test_examples.py`. I'll remove the extra prints.

Do you want the test to fail silently? I print the files so that if it fails the report will clearly show what triggered the fail.

---

Removed all the excess code. Kept using a variable for the assert so that a failure will be easier to understand.
Changes in next patch.

---

Try giving `assert  os.listdir(tempDirPath) == []` a try when the test is failing. I think you will find that it gives you all the information you want (pytest does a lot of magic with the assert statement).